### PR TITLE
TMA-905: do not print dynamic schedule param value if the param is set as secure

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/collect_dynamic_schedule_params.rb [hunk 1]
```diff
@@ -43,13 +43,15 @@ def call(params)
...
             File.open(data_source.realize(params), 'r:UTF-8')
           end
-          GoodData.logger.debug("Input data: #{input_data.read}")
 
           schedule_params = {}
 
           CSV.foreach(input_data, :headers => true, :return_headers => false, encoding: 'utf-8') do |row|
-            GoodData.logger.debug("Processing row: #{row}")
-            results << row.to_hash
+            safe_to_print_row = Hash[row.to_hash.map do |k, v|
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_dynamic_schedule_params.rb]: info: Contains iterators nested 2 deep (NestedIterators) (reek)

Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_dynamic_schedule_params.rb]: info: Has the variable name 'k' (UncommunicativeVariableName) (reek)

Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_dynamic_schedule_params.rb]: info: Has the variable name 'v' (UncommunicativeVariableName) (reek)

Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_dynamic_schedule_params.rb]: warning: Lint/UnusedBlockArgument: Unused block argument - `v`. If it's necessary, use `_` or `_v` as an argument name to indicate that it won't be used. (rubocop)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/571/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1395/ : FAILURE in 2m 54s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1665/ : SUCCESS in 2m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/572/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1396/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1666/ : SUCCESS in 1m 07s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by kubamahnert** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/575/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1398/ : SUCCESS in 1m 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1668/ : SUCCESS in 1m 12s

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/577/ : SUCCESS in 54s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1399/ : SUCCESS in 1m 24s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1669/ : SUCCESS in 1m 18s

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/726/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1544/ : SUCCESS in 37m 13s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/105/ : FAILURE in 2m 20s

File: spec/unit/actions/collect_dynamic_schedule_params_spec.rb [hunk 2]
```diff
@@ -52,5 +51,16 @@
       }
       expect(result[:params][:schedule_params]).to eq(expected)
     end
+
+    it 'does not print the one with param_secure' do
+      result = subject.class.call(params)
```
Comment (panjan) [inline spec/unit/actions/collect_dynamic_schedule_params_spec.rb]: We use the `GoodData::LCM2.run_action` method for testing actions now.

Comment (kubamahnert) [reply spec/unit/actions/collect_dynamic_schedule_params_spec.rb]: Fixed.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by kubamahnert** [comment]
@panjan as this logic does not rely on logic of other components besides gooddata-ruby, I suppose end to end testing would not bring anything more to the table.

**Comment by kubamahnert** [comment]
https://travis-ci.org/kubamahnert/gooddata-ruby/builds/448886670

**Comment by kubamahnert** [comment]
https://travis-ci.org/kubamahnert/gooddata-ruby/builds/449259534

**Comment by kubamahnert** [comment]
https://travis-ci.org/kubamahnert/gooddata-ruby/builds/449259534

**Pull request merged**